### PR TITLE
nixos/networking: do not copy /etc/rpc on non-glibc systems

### PR DESCRIPTION
**Pull request by tathougies**
###### Motivation for this change

Mandating using /etc/rpc from glibc means we may pull in an unnecessary dependency on non-glibc systems. This gets rid of that, allowing for alternate C libraries to provide their own in the future.

Scoped down version of #49218 

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by dtzWill** [comment]
See comment on previous PR: https://github.com/NixOS/nixpkgs/pull/49218#issuecomment-433943083 (probably should have posted it here instead, sorry).

Although currently AFAIK NFS doesn't work w/musl (since we upgraded it), but that shouldn't be especially hard to fix.  The major blocker for many things is librsvg -> rust, which mostly needs someone to create+host a musl-based rust bootstrap tarball -- possibly starting with those available in various musl-based distributions, and applying and fixes to rust that may be needed (not sure which these are for latest versions).

**Comment by matthewbauer** [comment]
Those commits I pushed might not be helpful if so definitely feel free to drop them.

> Although currently AFAIK NFS doesn't work w/musl (since we upgraded it), but that shouldn't be especially hard to fix. The major blocker for many things is librsvg -> rust, which mostly needs someone to create+host a musl-based rust bootstrap tarball -- possibly starting with those available in various musl-based distributions, and applying and fixes to rust that may be needed (not sure which these are for latest versions).

This should just effect native built Musl though right? The cross built stuff should be able to cross compile librsvg (someone was working on cross support for rust recently). Not to completely discourage the effort of getting Rust-based Musl bootstrap but it can become a maintenance burden over time. We have a couple of self-hosted compilers in Nixpkgs now and I'm hoping we can find a better solution than making double bootstrap tools for each one.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -9739,6 +9739,12 @@ with pkgs;
 
   libcCross = assert stdenv.targetPlatform != stdenv.buildPlatform; libcCrossChooser stdenv.targetPlatform.libc;
 
+  ldd = {
```
Comment (tathougies) [inline pkgs/top-level/all-packages.nix]: Wouldn't it just be easier to set this to `stdenv.hostPlatform.stdenv.cc.libc`? All libcs should provide ldd, if I'm not mistaken.

**Comment by tathougies** [comment]
@dtzWill I'll take a look at the branch. I'm using nixos as a base off of which to build an embedded ARM-based home server appliance, and I've mostly got everything I want working. Now, I'm contributing back the patches I think would make sense in mainline nixpkgs (should I work off your branch instead?). I've had to make substantial changes to nixpkgs, but I now have a musl-based ARM board that boots and gives me a getty. I've switched out systemd with runit to make it work of course, because systemd does not support musl, and I'm not going to fight that battle :)

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (matthewbauer) [reply pkgs/top-level/all-packages.nix]: Yeah that sounds like a good idea.

**Comment by matthewbauer** [comment]
@dtzWill what issues are you seeing on your branch? If you could collect together the safe commits I think we can get those merged (some of the systemd stuff might be too much).

**Comment by matthewbauer** [comment]
Merged in #49504

**Pull request closed without merging**